packages/worker/src/worker-options.ts

```diff
@@ -744,20 +796,21 @@ export function compileWorkerOptions(rawOpts: WorkerOptions, logger: Logger): Co
     logger.warn('maxCachedWorkflows must be either 0 (ie. cache is disabled) or greater than 1. Defaulting to 2.');
     opts.maxCachedWorkflows = 2;
   }
-  if (opts.maxCachedWorkflows > 0 && opts.maxConcurrentWorkflowTaskExecutions > opts.maxCachedWorkflows) {
-    logger.warn(
-      "maxConcurrentWorkflowTaskExecutions can't exceed maxCachedWorkflows (unless cache is disabled). Defaulting to maxCachedWorkflows."
-    );
-    opts.maxConcurrentWorkflowTaskExecutions = opts.maxCachedWorkflows;
-  }
-  if (opts.maxCachedWorkflows > 0 && opts.maxConcurrentWorkflowTaskExecutions < 2) {
-    logger.warn(
-      "maxConcurrentWorkflowTaskExecutions can't be lower than 2 if maxCachedWorkflows is non-zero. Defaulting to 2."
-    );
-    opts.maxConcurrentWorkflowTaskExecutions = 2;
-  }
+  // if (opts.maxCachedWorkflows > 0 && opts.maxConcurrentWorkflowTaskExecutions > opts.maxCachedWorkflows) {
+  //   logger.warn(
```

Comment:
This could technically stay, but would require some slightly annoying checks around looking to see if they're only using the "old" options. I can restore it if we think it's worth keeping.

Reply:
What will happen if they provide settings that don't respect these invariants? I think that should stay. However, it would be possible to move these to the corresponding if block in `addDefaultWorkerOptions`…

Reply:
They actually will get checked by Core anyway, if the supplier provides a total slots number which fixed size ones do

Reply:
I've added the check back, but, my inclination is to move all of this, including the checks for cached workflow settings, into Core since there's no reason to dupe this everywhere. I'll follow up on that in another Core PR.

Reply:
> my inclination is to move all of this, including the checks for cached workflow settings, into Core since there's no reason to dupe this everywhere. I'll follow up on that in another Core PR.

I'd also be in favor of avoiding that kind of dupe, but I must point out these checks were _intentionally_ added here (knowing they were dup'ed) because that allowed better warning/error reporting, and made it easier to handle default values (after all, these checks are pretty much linked to the ones just above that will determine a default value for `maxCachedWorkflows`, number of pollers, etc). So I'm not against moving this out to Core, but that might be trickier to do properly than it seems at first.